aesd-char-driver/main.c

```diff
@@ -175,12 +187,143 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 
     return count;
 }
+
+// Implement fixed-size llseek() discussed in assignment course video
+loff_t aesd_llseek(struct file *filp, loff_t off, int whence) 
+{
+    ssize_t retval = -EFAULT;
+    struct aesd_dev *dev = filp->private_data;
+    struct aesd_buffer_entry *entry;
+    int i = 0, circ_buf_size = 0;
+
+    PDEBUG("llseek type %d with offset %lld", whence, off);
+
+    // tally current circular buffer size
+    AESD_CIRCULAR_BUFFER_FOREACH(entry, &dev->circ_buf, i) {
+        if(entry->buffptr != NULL) {
+            circ_buf_size += entry->size;
+        }
+    }
+
+    // lock cbuffer access
+    if(mutex_lock_interruptible(&dev->mut) != 0) {
```

Comment:
@betai97 you don't need locking for fixed_size_llseek, only for the operation on the circular buffer below.